x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleRunner.java

```diff
@@ -303,6 +303,12 @@ public void onResponse(Boolean complete) {
                                 // index since it will be... deleted.
                                 registerDeleteOperation(indexMetadata);
                             }
+                        } else {
+                            // All steps *should* return true for complete, or invoke listener.onFailure
```

Comment:
I wonder if it'd make sense to convert the listener from `ActionListener<Boolean>` to `ActionListener<Void>` so we erase the `false` branch completely (this would entail changing the signature of `performAction`) ? I think the method signature would be clearer too as now it indicates a tri-state possibility for `onResponse` (null, true, false) but the listener should actually only promote only one state for success (null in case we make it `ActionListener<Void>`).

What do you think?

Reply:
@andreidan I think that sounds good, are you okay if I do that in a follow-up only for the 7.x and 8.0 branches, or would you prefer I do it in this change?

Reply:
Follow-up sounds great. Thanks Lee